apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/BigQueryInsertRowErrorsTests.cs

```diff
@@ -0,0 +1,162 @@
+                            Reason = "Very bad reason",
+                            Message = "A very bad error occured"
+                        }
+                    }
+                }
+            };
+
+            SingleError expected = new SingleError
+            {
+                Reason = "Very bad reason",
+                Message = "Error in row unknown. A very bad error occured"
+            };
+
+            // This is allowed in case there's no information (index)
+            // to associate errors to a row.
+            BigQueryInsertRowErrors insertRowErrors = new BigQueryInsertRowErrors(null, errors);
+            Assert.Single(insertRowErrors);
+            Assert.Contains(expected, insertRowErrors, new SingleErrorEqualityComparer());
+        }
+
+        [Fact]
+        public void GetEnumerator()
+        {
+            List<InsertErrorsData> errors = new List<InsertErrorsData>
+            {
+                new InsertErrorsData
+                {
+                    Index = 1,
+                    Errors = new List<ErrorProto>
+                    {
+                        new ErrorProto
+                        {
+                            Reason = "Reason 1",
+                            Message = "A very bad error 1 occured",
+                            Location = "field_1"
+                        },
+                        new ErrorProto
+                        {
+                            Reason = "Reason 2",
+                            Location = "field_2"
+                        }
+                    }
+                },
+                new InsertErrorsData { },
+                new InsertErrorsData
+                {
+                    Index = 1,
+                    Errors = new List<ErrorProto>
+                    {
+                        new ErrorProto
+                        {
+                            Reason = "Reason 3",
+                            Message = "A very bad error 3 occured",
+                            Location = "field_3"
+                        }
+                    }
+                },
+                null
+            };
+
+            IList<SingleError> expected = new List<SingleError>
+            {
+                new SingleError
+                {
+                    Reason = "Reason 1",
+                    Message = "Error in row 1. A very bad error 1 occured",
+                    Location = "field_1"
+                },
+                new SingleError
+                {
+                    Reason = "Reason 2",
+                    Message = "Error in row 1. ",
+                    Location = "field_2"
+                },
+                new SingleError
+                {
+                    Reason = "Reason 3",
+                    Message = "Error in row 1. A very bad error 3 occured",
+                    Location = "field_3"
+                }
+            };
+
+            // This is allowed in case there's no information (index)
```

Comment:
We've passed null as the first constructor argument in all the tests. Shouldn't we have some tests for non-null arguments?